radicle-node/src/worker.rs

```diff
@@ -100,7 +104,6 @@ pub struct TaskResult<G: Signer + Ecdh> {
 
 /// A worker that replicates git objects.
 struct Worker<G: Signer + Ecdh> {
-    local: PublicKey,
```

Comment:
* If we force-fetch `rad/id`, then yes, the refs won't be rejected.
* If we don't use `--atomic`, a rejected ref won't fail the fetch

These are the two reasons I can think of, but I'm not sure they are good reasons, ie. I think we may not want to force `rad/id` and we may want to use `--atomic`.

Reply:
`--atomic` only exists using the `git` command line though iirc, but in our final step we're using `git2` for local transfer

Reply:
True yes, it only applies to the fetch into the staging area.

Reply:
Right, so this is a good reason _not_ to include `rad/id` in the sigrefs

Reply:
Oh interesting :thinking: -- this is something we discussed on zulip right? And then thought we'd leave it as it "doesn't hurt".

The only thing is, I think parts of the code expect that *all* refs are signed in sigrefs. It makes verification simpler, with two rules: (1) all refs in sigrefs must exist, and (2) all refs that exist must be in sigrefs.

Removing rad/id would add an annoying exception to that rule.

Another option is to leave rad-id in there, but special case it to not allow force-push.

But this makes me wonder: is there any harm in force-pushing rad/id? I guess we'd have to change the "longest branch" logic.. :thinking:

Reply:
Hmmmm I guess it can't go wrong because we've fetched `rad/id` anyway, so it should essentially be a no-op because it would be hard (and useless) to have a different `sigrefs` entry.

Reply:
So are you proposing to leave it in sigrefs because that'll fail on verification if it is force-pushed?

Reply:
Ah I see, so we fetch without force, and we keep rad/id inside sigrefs.